checkPropTypes.js

```diff
@@ -25,7 +25,7 @@ if (process.env.NODE_ENV !== 'production') {
  * @param {?Function} getStack Returns the component stack.
  * @private
  */
-function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
+function checkPropTypes(typeSpecs, values, location, componentName, getStack, warningLogger = null, shouldLogAllTypeFailures = false) {
```

Comment:
```suggestion
function checkPropTypes(typeSpecs, values, location, componentName, getStack, warningLogger = warning, shouldLogAllTypeFailures = false) {
```

Reply:
This boolean argument seems unnecessary to me, since you can encode that logic in your custom logger function.

Reply:
@ljharb , its been a year since the PR is made , if you say i can make a new updated PR without any merge conflicts

Reply:
@kushthedude the time elapsed shouldn’t be relevant; please do *not* make a separate PR.

Reply:
Removed the flag since it's now irrelevant (since https://github.com/facebook/prop-types/commit/40af72f41a404f081f87e5f160bb70c60401e587)